vue/components/ui/molecules/image-input/image-input.vue

```diff
@@ -0,0 +1,188 @@
+    border-radius: 6px;
+    cursor: pointer;
+}
+
+.image-input:hover {
+    border-color: #1d1d1d;
+}
+
+.image-input.dragging {
+    opacity: 0.3;
+}
+
+.image-input > .image-input-content {
+    box-sizing: border-box;
+    cursor: pointer;
+    display: flex;
+    flex-direction: column;
+    height: 100%;
+    padding: 10px 10px 10px 10px;
+    width: 100%;
+}
+
+.image-input > .image-input-content > .image-container {
+    align-items: center;
+    cursor: initial;
+    display: flex;
+    overflow: hidden;
+    padding: 5px 0px 5px 0px;
+    white-space: nowrap;
+}
+
+.image-input > .image-input-content > .image-container > .remove-button {
+    cursor: pointer;
+    display: none;
+    margin-right: 10px;
+}
+
+.image-input > .image-input-content > .image-container:hover > .remove-button {
+    display: inline-block;
+}
+
+.image-input > .image-input-content > .image-container > .image-preview {
+    height: 30px;
+    vertical-align: top;
+    width: 30px;
+}
+
+.image-input > .image-input-content > .image-container > .image-preview ::v-deep img {
+    object-fit: contain;
+}
+
+.image-input > .image-input-content > .image-container > .name {
+    display: inline-block;
+    line-height: 30px;
+    margin: 0px 10px;
+    width: 100%;
+}
+</style>
+
+<script>
+export const ImageInput = {
+    name: "image-input",
+    props: {
+        /**
+         * The initial set of images to be uploaded.
+         * An image should be an object with data property
+         * having the content of the image in base64.
+         */
+        images: {
+            type: Array,
+            default: []
+        },
+        /**
+         * Support for multiple images, this allowing the user
+         * to select and upload more than one image.
+         */
+        multiple: {
+            type: Boolean,
+            defsault: true
+        }
+    },
+    computed: {
+        hasImages() {
+            return this.imagesData.length > 0;
+        }
+    },
+    data: function() {
+        return {
+            imagesData: this.images,
+            visibleLightbox: null
+        };
+    },
+    watch: {
+        imagesData(value) {
+            this.$emit("update:images", value);
+        }
+    },
+    methods: {
+        loadFiles(files) {
+            if (this.multiple) {
+                Object.values(files).forEach(file => this.loadFile(file));
+            } else {
+                this.imagesData = [];
+                this.loadFile(files[0]);
+            }
+        },
+        onClickFileRemove(index) {
+            this.imagesData.splice(index, 1);
+        },
+        onClickLightbox(index) {
+            this.visibleLightbox = index;
+        },
+        onCloseLightbox() {
+            this.visibleLightbox = null;
+        },
+        async loadFile(file) {
+            const data = await new Promise((resolve, reject) => {
+                const reader = new FileReader();
+                reader.readAsDataURL(file);

```

Comment:
Shouldn't this go after callback registration? I know it still works, but doesn't seem 100% error-proof